cloud_experts_tutorials/rosa-mobb-aws-secret-manager.adoc

```diff
@@ -0,0 +1,333 @@
+// ---
+// date: '2023-05-25'
+// title: Using AWS Secrets Manager CSI on Red Hat OpenShift on AWS with STS
+// tags: ["AWS", "ROSA"]
+// authors:
+//   - Paul Czarkowski
+//   - Chris Kang
+// ---
+
+The AWS Secrets and Configuration Provider (ASCP) provides a way to expose AWS Secrets as Kubernetes storage volumes. With the ASCP, you can store and manage your secrets in Secrets Manager and then retrieve them through your workloads running on ROSA.
+
+This is made even easier and more secure through the use of AWS STS and Kubernetes PodIdentity.
+
+[id="rosa-mobb-aws-secret-manager-prerequisites"]
+== Prerequisites
+
+* A ROSA cluster deployed with STS
+* Helm 3
+* aws CLI
+* oc CLI
+* jq
+
+[id="rosa-mobb-aws-secret-manager-preparing-environment"]
+== Preparing Environment
+
+. Validate that your cluster has STS:
++
+[source,terminal]
+----
+$ oc get authentication.config.openshift.io cluster -o json \
+  | jq .spec.serviceAccountIssuer
+----
++
+You should see something like the following, if not you should not proceed, instead look to the link:https://docs.openshift.com/rosa/rosa_getting_started_sts/rosa_creating_a_cluster_with_sts/rosa-sts-creating-a-cluster-quickly.html[Red Hat documentation on creating an STS cluster].
++
+[source,terminal]
+----
+"https://xxxxx.cloudfront.net/xxxxx"
+----
+
+. Set SecurityContextConstraints to allow the CSI driver to run:
++
+[source,terminal]
+----
+$ oc new-project csi-secrets-store
+$ oc adm policy add-scc-to-user privileged \
+    system:serviceaccount:csi-secrets-store:secrets-store-csi-driver
+$ oc adm policy add-scc-to-user privileged \
+    system:serviceaccount:csi-secrets-store:csi-secrets-store-provider-aws
+----
+
+. Create some environment variables to refer to later:
++
+[source,terminal]
+----
+$ export REGION=$(oc get infrastructure cluster -o=jsonpath="{.status.platformStatus.aws.region}")
+$ export OIDC_ENDPOINT=$(oc get authentication.config.openshift.io cluster \
+   -o jsonpath='{.spec.serviceAccountIssuer}' | sed  's|^https://||')
+$ export AWS_ACCOUNT_ID=`aws sts get-caller-identity --query Account --output text`
+$ export AWS_PAGER=""
+----
+
+[id="rosa-mobb-aws-secret-manager-deply-aws-secrets"]
+== Deploy the AWS Secrets and Configuration Provider
+
+. Use Helm to register the secrets store CSI driver:
++
+[source,terminal]
+----
+$ helm repo add secrets-store-csi-driver \
+    https://kubernetes-sigs.github.io/secrets-store-csi-driver/charts
+----
+
+. Update your Helm repositories:
++
+[source,terminal]
+----
+$ helm repo update
+----
+
+. Install the secrets store CSI driver:
++
+[source,terminal]
+----
+$ helm upgrade --install -n csi-secrets-store \
+    csi-secrets-store-driver secrets-store-csi-driver/secrets-store-csi-driver
+----
+
+. Deploy the AWS provider:
++
+[source,terminal]
+----
+$ oc -n csi-secrets-store apply -f \
+    https://raw.githubusercontent.com/rh-mobb/documentation/main/content/misc/secrets-store-csi/aws-provider-installer.yaml
+----
+
```

Comment:
Insert a new step here:

Label the Secrets Store CSI Driver to allow use with the restricted pod security profile:

```
$ oc label csidriver.storage.k8s.io/secrets-store.csi.k8s.io security.openshift.io/csi-ephemeral-volume-profile=restricted
```